### ISSUE TITLE
Failed front-end build deployed anyway

### ISSUE THREAD

**Issue by ctm** [ctm/mb2-doc#1209]
Fix the deploy script to not attempt to deploy a failed web build.

This just happened and I was distracted and didn't notice. FWIW, I didn't think that could happen since we have `errexit` set in that script. I guess the failure in question (which was due to some code for `late-183` accidentally slipping into `master`) didn't exit with a non-zero error code. 

Before fixing `master`, I saved a `master-fail` branch, so this is trivially reproducible.

**Comment by ctm** [comment]
`yarn build` failed but returned a 0 exit code. I briefly explored upgrading to yarn 3, but that looks like a **very bad idea** right now, so it makes sense to see if I can detect a failure-to-compile breakage and insert that into `deploy`. That alone won't guarantee that all breakages are fixed, but that's the one that bit us last night (and has also bitten us in the past, I'm embarrassed to admit).

The file `dist/mb2-web.wasm` is not created during a failed build, so if I make deploy remove that file before the `yarn build` and test after it, that will fix the problem we had last night.

**Comment by ctm** [comment]
I've added that test, verified that it catches a failed build and gone on and done a deploy with it. Before I added the test I did a minor refactor and extracted the common front-end building code from `deploy` and `build_and_send` so that the fix only needs to be in one place. There's still too much duplicated code between everything in bin (`deploy`, `build_and_send`, `revert_to_tag`, `sent_deploy` and `local_docker`), but I'm going to let those sleeping dogs lie, for now.
